### PR TITLE
Add headers while downloading file as user

### PR DESCRIPTION
**Pull request by dpakach**
## Description

The function to  download file as a user takes user, filename and password  only.
But we can add headers also to add extra options on the download

## Related Issue

- Required for https://github.com/owncloud/firewall/issues/539

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Review by phil-davis — APPROVED** [review]
@dpakach I guess you plan to use this in code somewhere?

**Review by paurakhsharma — APPROVED** [review]

**Review by individual-it — APPROVED** [review]

**Comment by dpakach** [comment]
Backport in #33751

**Comment by dpakach** [comment]
@phil-davis yes I need this in firewall app

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -957,14 +957,14 @@ public function userUsingPasswordDownloadsTheFileUsingTheAPI(
 	 * @return void
 	 */
 	public function downloadFileAsUserUsingPassword(
-		$user, $fileName, $password = null
+		$user, $fileName, $password = null, $headers = []
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]: PHPdoc for new parameter is missing
https://drone.owncloud.com/owncloud/core/13161/100

Comment (phil-davis) [reply tests/acceptance/features/bootstrap/WebDav.php]: I fixed, ssquashed and pushed.

**Pull request merged**